python/semantic_kernel/data/const.py

```diff
@@ -1,21 +1,93 @@
 # Copyright (c) Microsoft. All rights reserved.
 
-
 from enum import Enum
+from typing import Final
+
+DEFAULT_FUNCTION_NAME: Final[str] = "search"
+DEFAULT_DESCRIPTION: Final[str] = (
+    "Perform a search for content related to the specified query and return string results"
+)
 
 
 class IndexKind(str, Enum):
-    """Index kinds for similarity search."""
+    """Index kinds for similarity search.
+
+    HNSW
+        Hierarchical Navigable Small World which performs an approximate nearest neighbor (ANN) search.
+        Lower accuracy than exhaustive k nearest neighbor, but faster and more efficient.
+
+    Flat
+        Does a brute force search to find the nearest neighbors.
+        Calculates the distances between all pairs of data points, so has a linear time complexity,
+        that grows directly proportional to the number of points.
+        Also referred to as exhaustive k nearest neighbor in some databases.
+        High recall accuracy, but slower and more expensive than HNSW.
+        Better with smaller datasets.
+
+    IVF Flat
+        Inverted File with Flat Compression.
+        Designed to enhance search efficiency by narrowing the search area
+        through the use of neighbor partitions or clusters.
+        Also referred to as approximate nearest neighbor (ANN) search.
+
+    Disk ANN
+        Disk-based Approximate Nearest Neighbor algorithm designed for efficiently searching
+        for approximate nearest neighbors (ANN) in high-dimensional spaces.
+        The primary focus of DiskANN is to handle large-scale datasets that cannot fit entirely
+        into memory, leveraging disk storage to store the data while maintaining fast search times.
+
+    Quantized Flat
+        Index that compresses vectors using DiskANN-based quantization methods for better efficiency in the kNN search.
+
+    Dynamic
+        Dynamic index allows to automatically switch from FLAT to HNSW indexes.
+
+    """
 
     HNSW = "hnsw"
     FLAT = "flat"
+    IVF_FLAT = "ivf_flat"
+    DISK_ANN = "disk_ann"
+    QUANTIZED_FLAT = "quantized_flat"
+    DYNAMIC = "dynamic"
 
 
 class DistanceFunction(str, Enum):
```

Comment:
Any plans to support the Minkowski distance metric?

Reply:
@westey-m have you seen that one in any of the current ones?